course/models.py

```diff
@@ -662,8 +662,8 @@ def get_visible(self, user=None):
             )
         if not user.is_superuser:
             return self.filter(
-                get_course_visibility_filter(user, 'course_instance__'),
-                opening_time__lte=timezone.now(),
+                Q(course_instance__visible_to_students=True, opening_time__lte=timezone.now())
```

Comment:
These Q + visibility_fitler parts are in different order in these two locations. Might be easier to read of they would have the same order always. Doesn't affect the logic though.